File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        maxArea = 0
        while(l < r):
            length = r - l
            width = min(height[l], height[r])
            area = length * width
            maxArea = max(area, maxArea)
            if(height[l] < height[r]):
                l +=1 
            else:
                r -= 1
        return maxArea 
        

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number} 
 */
var maxArea = function(height) {
    let l = 0, r = height.length-1, maxwater = 0

    while(l < r){
        let h = Math.min(height[l], height[r])
        let b = r-l

        maxwater = Math.max(maxwater, h*b)

        if(height[l] < height[r]){
            l++
        }else{
            r--
        }
    }

    return maxwater
};